app/Http/Controllers/LearningController.php

```diff
@@ -159,10 +162,34 @@ public function updateProgress(Request $request, $id)
             $progress->save();
         }
 
+        $mytime = Carbon::now();
+        $progress->updated_at = $mytime->toDateTimeString();
+        $progress->save();
+
         return response()->json([
             'success' => true,
             'message' => __('Progress updated'),
             'progress' => $progress
         ]);
     }
+
+    public function checkHasUpdated($id)
+    {
+        $user_id = Auth::id();
+
+        $lessons = Lesson::select('title')
+            ->whereHas('progress', function ($query) use ($user_id) {
+                $query->where('user_id', $user_id)
+                    ->whereRaw('progress.updated_at < lessons.updated_at');
+            })
+            ->where('course_id', $id)
+            ->get();
+        $warning_messages = [];
+
+        foreach ($lessons as $lesson) {
+            $warning_messages[] = __('Lesson ') . ($lesson->title) . __(' has been updated');
```

Comment:
https://laravel.com/docs/10.x/localization#replacing-parameters-in-translation-strings